src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,136 @@
+        return out.toString();
+    }
+
+    String sayBye() {
+        return "\tBye. Hope to see you again soon!";
+    }
+
+    String[] addTask(String input) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        String[] split = input.split(" ", 2);
+        String taskType = split[0];
+        try {
+            switch (taskType) {
+            case "t": //fallthrough
+            case "todo": {
+                if (split[1].equals("")) {
+                    DukeException.rethrow("ToDoException");
+                }
+                Task task = new ToDo(split[1]);
+                taskList.add(task);
+                return new String[]{String.valueOf(taskList.size()), String.format("added: %s", task)};
+            }
+            case "d": {
+                DateTimeFormatter altFormatter = DateTimeFormatter.ofPattern("EEE, d MMM yyyy hh:mma");
+                String[] s = split[1].split(" by: ");
+                System.out.println(s[0]);
+                LocalDateTime byTime = LocalDateTime.parse(s[1], altFormatter);
+                Task task = new Deadline(s[0], byTime);
+                taskList.add(task);
+                return new String[]{String.valueOf(taskList.size()), String.format("added: %s", task)};
+            }
+            case "deadline": {
+                String[] s = split[1].split(" /by ");
+                System.out.println(s[0]);
+                LocalDateTime byTime = LocalDateTime.parse(s[1], formatter);
+                Task task = new Deadline(s[0], byTime);
+                taskList.add(task);
+                return new String[]{String.valueOf(taskList.size()), String.format("added: %s", task)};
+            }
+            case "e": {
+                DateTimeFormatter altFormatter = DateTimeFormatter.ofPattern("EEE, d MMM yyyy hh:mma");
+                String[] s = split[1].split(" from: ", 2);
+                String taskDescription = s[0];
+                s = s[1].split(" to: ");
+                LocalDateTime fromTime = LocalDateTime.parse(s[0], altFormatter);
+                LocalDateTime toTime = LocalDateTime.parse(s[1], altFormatter);
+                Task task = new Event(taskDescription, fromTime, toTime);
+                taskList.add(task);
+                return new String[]{String.valueOf(taskList.size()), String.format("added: %s", task)};
+            }
+            case "event": {
+                String[] s = split[1].split(" /from ", 2);
+                String taskDescription = s[0];
+                s = s[1].split(" /to ");
+                LocalDateTime fromTime = LocalDateTime.parse(s[0], formatter);
+                LocalDateTime toTime = LocalDateTime.parse(s[1], formatter);
+                Task task = new Event(taskDescription, fromTime, toTime);
+                taskList.add(task);
+                return new String[]{String.valueOf(taskList.size()), String.format("added: %s", task)};
+            }
```

Comment:
Any reason why you use one method to handle addTask inputs from both the CLI and the saved file?